ftw/zipextract/browser/zipextract_view.py

```diff
@@ -0,0 +1,71 @@
+from Products.Five.browser import BrowserView
+from ftw.zipextract.zipextracter import ZipExtracter
+from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
+from plone.i18n.normalizer import idnormalizer
+import os
+from z3c.form import form
+
+class ZipExtractView(form.Form):
+
+    template = ViewPageTemplateFile('templates/zipextract.pt')
+
+    def __call__(self):
+        self.zipextracter = ZipExtracter(self.context)
+        self.build_tree_html()
```

Comment:
This does not need to be done when the form is submitted, right?

Reply:
No indeed... I wanted to ask about this but forgot. How do I do this properly? when the form is submitted, it seems that a new `ZipExtractView` is created and then the `__call__` method gets called...

Reply:
Ok nevermind, I misunderstood. `self.zipextracter` gets generated anew when the form is submitted, but that cannot be avoided. The html tree is now handled by the template so that is not an issue anymore.